x-pack/metricbeat/module/gcp/vertexai_logs/vertexai_logs.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	parts := strings.Split(c.TableID, ".")
+	if len(parts) != 3 {
+		return fmt.Errorf("table_id must be in format 'project_id.dataset_id.table_name', got: %s", c.TableID)
+	}
+
+	if c.TimeLookbackHours < 0 {
+		return errors.New("time_lookback_hours must be non-negative")
+	}
+
+	return nil
+}
+
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	m := &MetricSet{
+		BaseMetricSet: base,
+		logger:        base.Logger().Named(metricsetName),
+	}
+
+	if err := base.Module().UnpackConfig(&m.config); err != nil {
+		return nil, fmt.Errorf("unpack vertexai_logs config failed: %w", err)
+	}
+
+	// Set defaults
+	if m.config.TimeLookbackHours == 0 {
+		m.config.TimeLookbackHours = 1 // Default: 1 hour
+	}
+
+	m.logger.Debugf("metricset config: project_id=%s, dataset_id=%s, table_name=%s, time_lookback=%dh",
+		m.config.ProjectID, getDatasetID(m.config.TableID), getTableName(m.config.TableID),
+		m.config.TimeLookbackHours)
+	return m, nil
+}
+
+func getDatasetID(tableID string) string {
+	parts := strings.Split(tableID, ".")
+	if len(parts) >= 3 {
+		return parts[1]
+	}
+	return ""
+}
+
+func getTableName(tableID string) string {
+	parts := strings.Split(tableID, ".")
+	if len(parts) >= 3 {
+		return parts[2]
+	}
+	return ""
+}
+
+func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+	var opt []option.ClientOption
+	if m.config.CredentialsFilePath != "" && m.config.CredentialsJSON != "" {
+		return errors.New("both credentials_file_path and credentials_json specified, you must use only one of them")
+	} else if m.config.CredentialsFilePath != "" {
+		opt = []option.ClientOption{option.WithCredentialsFile(m.config.CredentialsFilePath)}
+	} else if m.config.CredentialsJSON != "" {
+		opt = []option.ClientOption{option.WithCredentialsJSON([]byte(m.config.CredentialsJSON))}
+	} else {
+		return errors.New("no credentials_file_path or credentials_json specified")
+	}
+
+	client, err := bigquery.NewClient(ctx, m.config.ProjectID, opt...)
+	if err != nil {
+		return fmt.Errorf("error creating bigquery client: %w", err)
+	}
+	defer client.Close()
+
+	datasetID := getDatasetID(m.config.TableID)
+	dataset := client.Dataset(datasetID)
+	meta, err := dataset.Metadata(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting dataset metadata: %w", err)
+	}
+
+	events, err := m.queryVertexAILogs(ctx, client, meta.Location)
+	if err != nil {
+		return fmt.Errorf("queryVertexAILogs failed: %w", err)
+	}
+
+	m.logger.Debugf("Total %d events created for vertexai_logs", len(events))
+	for _, event := range events {
+		reporter.Event(event)
+	}
+
+	// Update watermark with latest logging_time from events
+	m.updateLastLoggingTime(events)
+
+	return nil
+}
+
+func (m *MetricSet) queryVertexAILogs(ctx context.Context, client *bigquery.Client, location string) ([]mb.Event, error) {
+	query := m.generateQuery()
+	m.logger.Debug("bigquery query = ", query)
+
+	q := client.Query(query)
+	q.Location = location
```

Comment:
not a blocker: i don't think this is strictly necessary - it's already part of the generated query (in the `FROM`).

if we remove this we can also remove the parsing functions for tablename and  location, which simpliies things a little.